src/rust/engine/process_execution/sandboxer/src/lib.rs

```diff
@@ -208,3 +211,87 @@ impl SandboxerGrpc for SandboxerGrpcImpl {
         ret
     }
 }
+
+// The client that sends requests to the sandboxer service.
+#[derive(Clone, Debug)]
+pub struct SandboxerClient {
+    grpc_client: SandboxerGrpcClient<Channel>,
+}
+
+impl SandboxerClient {
+    pub async fn connect(socket_path: &Path) -> Result<Self, String> {
+        let socket_path_buf = socket_path.to_path_buf();
+        let channel = Endpoint::try_from("http://[::]:50051")
```

Comment:
This "endpoint" is an artifact of how UDS support got tacked on in tonic. 50051 is the official gRPC port, but of course this does not signify since we're not using TCP/IP.

Reply:
Tonic's `uds` example shows how to use a `unix://` scheme. https://github.com/hyperium/tonic/blob/50253f1cb236feca3061488d542e37cf60ba4f65/examples/src/uds/client_standard.rs#L16.

Although to your point, the example which uses `connect_with_conector` does have to provide the dummy port information. Maybe link to example?

https://github.com/hyperium/tonic/blob/50253f1cb236feca3061488d542e37cf60ba4f65/examples/src/uds/client_with_connector.rs#L21

Reply:
This requires a parseable URI, but it doesn't actually matter what it is. It doesn't require the port, per se, I can remove that and link to the example.

Reply:
PTAL